src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -287,4 +578,72 @@ void assertClusterState(final ArgumentCaptor<ClusterStateUpdateTask> clusterStat
         assertNotNull(securityMetadata.configuration());
     }
 
+    private static void setupIndexResponseForDefaultConfig(IndexResponse indexResponse) {
+        OngoingStubbing<String> stubbingOperation = when(indexResponse.getId());
+
+        for (String configId : getConfigurationIdsInOrder()) {
+            stubbingOperation = stubbingOperation.thenReturn(configId);
+        }
+    }
+
+    private static PlainActionFuture<IndexResponse> getIndexResponsePlainActionFuture(IndexResponse indexResponse) {
+        return new PlainActionFuture<>() {
+            @Override
+            public IndexResponse actionGet() {
+                return indexResponse;
+            }
+        };
+    }
+
+    private static PlainActionFuture<ClusterHealthResponse> getClusterHealthResponsePlainActionFuture(
+        ClusterHealthResponse clusterHealthResponse
+    ) {
+        return new PlainActionFuture<>() {
+            @Override
+            public ClusterHealthResponse actionGet() {
+                return clusterHealthResponse;
+            }
+        };
+    }
+
+    private static PlainActionFuture<CreateIndexResponse> getCreateIndexResponsePlainActionFuture(CreateIndexResponse createIndexResponse) {
+        return new PlainActionFuture<>() {
+            @Override
+            public CreateIndexResponse actionGet() {
+                return createIndexResponse;
+            }
+        };
+    }
+
+    private static void setupObjectMapperInjectables() {
+        InjectableValues.Std injectableValues = new InjectableValues.Std();
+        injectableValues.addValue(Settings.class, Settings.EMPTY);
+        DefaultObjectMapper.inject(injectableValues);
+    }
+
+    // The order is important here as it matches order in initOnNodeStart()
+    private static String[] getConfigurationIdsInOrder() {
+        return new String[] {
```

Comment:
Should we consider getting these with something like `CType.values().stream().map(t -> t.toLCString()).toArray();`

Reply:
Currently, the order is not same during upload initOnNodeStart() to match order returned by Ctype.values(). For the mock to succeed we need exact type to be returned.  https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/support/ConfigHelper.java#L101